### ISSUE TITLE
'C:\Users\pc' is not recognized as an internal or external command, operable program or batch file. exit status 1 Error compiling for board Generic PY32F002A Series.

### ISSUE THREAD

**Issue by trimarco232** [py32duino/Arduino-PY32#10]
### 描述一下这个bug / Describe the bug

trying to compile leads immediately to the above error message
it compiles for AVR , STM32duino , ESP32 , lgt8f , aso.

### 复现步骤 / To Reproduce

maybe 'C:\Users\pc' is the beginning of  'C:\Users\pc de marco\' 
where 'pc de marco' is my user name
maybe there is a parser or something witch can't handle the space character

### 如果正常，应该是什么样 / Expected behavior

aa

### 截图 / Screenshots

aa

### 日志 / Logs

aa

### 系统 / System

windows 10

### PACK包版本 / Version

1.8.19 (same bug with 2.3.0)

### 验证

- [X] 检查过该问题，之前没有人提过 / Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] 提供了最小可复现工程或详细的复现步骤，确保开发者可以复现 / The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.
- [X] 已经提供了完整的报错信息、日志、截图，没有经过删减。

**Comment by HalfSweet** [comment]
Yes...I don't know what happens to gcc when linking, but it reports an error at the link stage when there are spaces or qi ta non-ascii characters in the compiler path.

I'll try to fix this as much as I can
